src/neo/Utility.cs

```diff
@@ -6,6 +8,32 @@ namespace Neo
 {
     public static class Utility
     {
+        internal class Logger : ReceiveActor
+        {
+            public Logger()
+            {
+                Receive<Debug>(e => OnLog(LogLevel.Debug, e.ToString()));
+                Receive<Info>(e => OnLog(LogLevel.Info, e.ToString()));
+                Receive<Warning>(e => OnLog(LogLevel.Warning, e.ToString()));
+                Receive<Error>(e => OnLog(LogLevel.Error, e.ToString()));
+                Receive<InitializeLogger>(_ => Init(Sender));
+            }
+
+            private void Init(IActorRef sender)
+            {
+                sender.Tell(new LoggerInitialized());
+            }
+
+            private void OnLog(LogLevel level, string message)
+            {
+                Log<NeoSystem>(level, message);
+            }
+        }
+
+        public delegate void delOnLog(string source, LogLevel level, string message);
+
+        public static event delOnLog OnLog;
```

Comment:
Is this event used in anywhere?

Reply:
It could be used for merge the errors with the `error. log` file in neo-cli

Reply:
You can register cli as a plugin. Or just create a plugin in cli.